### DISCUSSION TITLE
MeshRestriction in Multiphenicsx

### DISCUSSION THREAD

**Discussion by Sidaty1** [multiphenics/multiphenicsx#5] [Q&A] [answered]
Hi Guys, So far in Multiphenics I was able to use `multiphenics.MeshRestriction` in order to get the subdomain of interest, I would like to know what would be the equivalent funciton in Multiphenicsx, I have checked some examples in the demos repo but none seems to answer the question. Could you please help with that. 

PS: please find below  an example of what I would like to translate to Multiphenicsx. 

`mesh_restricted = mph.MeshRestriction(mesh, "sub.rtc.xml") `

Where the file "sub.rtc.xml" contains the cells, facets and vertices of my desired submesh. 

Thanks, 

Sidaty,

**Comment by francesco-ballarin** [comment, marked as answer]
The concept of MeshRestriction is not required anymore in multiphenicsx, due to the way the new FEniCSx handles degrees of freedom.

The closest concept we have in multiphenicsx is DofMapRestriction, which already operates on DOFs, rather than mesh entities. Please compare `tutorials/03_lagrange_multipliers/tutorial_lagrange_multipliers_linear.ipynb` to the corresponding multiphenics tutorial.

**Comment by Sidaty1** [comment]
Hi Francesco, 

Thank you for the help, Indeed, the Lagrange multipliers linear example was very helpful.  I've managed solve my problem by implementing another version of `locate_entities()` in dolfinx more adapted to my problem. 

Best, 

Sidaty,

**Reply by francesco-ballarin** [reply]
Great. Feel free to click on the "Mark as answer" button in my answer above, so that users will see that answer highlighted for future reference

**Reply by Sidaty1** [reply]
Done ;)
